internal/tfimport/importer/resources.go

```diff
@@ -28,3 +49,91 @@ func fromConfigValues(key string, cvs ...ProviderConfigMap) (interface{}, error)
 	}
 	return nil, fmt.Errorf("could not find key %q in resource change or provider config", key)
 }
+
+// Small helper functions to DRY the fromUser* functions.
+func showPrompt(fieldName, prompt string) {
+	log.Printf("Could not determine %q automatically\n", fieldName)
+	log.Println(prompt)
+}
+func parseUserVal(fieldName, val string, err error) (string, error) {
+	if val == "" {
+		ie := &InsufficientInfoErr{missingFields: []string{fieldName}}
+		if err != nil {
+			ie.msg = err.Error()
+		}
+		return "", ie
+	}
+	return val, err
+}
+
+// fromUser asks the user for the value, without providing choices.
+func fromUser(in io.Reader, fieldName string, prompt string) (val string, err error) {
+	showPrompt(fieldName, prompt)
+	val, err = userValue(in)
+	return parseUserVal(fieldName, val, err)
+}
+
+// fromUser asks the user to choose a value from the available choices.
+func fromUserChoices(in io.Reader, fieldName string, prompt string, choices []string) (val string, err error) {
```

Comment:
I am still a bit confused by this, what resource is using this? random id only seem to need fromUser + userValue.

Reply:
None yet, but I think some might. I'd like to keep it for now since I already wrote the code, and if I go through several non-trivial resources and none of them require it, I'll remove it.

Reply:
Removed it, but I have it in a separate branch. I can pull it back if needed in the future.